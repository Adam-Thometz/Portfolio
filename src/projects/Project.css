.Project {
  width: 85%;
  margin: 50px auto;
  border-radius: 20px;
  padding: 20px;
}

.Project h2 {
  text-align: center;
}

.Project-info-img-container {
  display: flex;
}

.Project-award {
  height: 50px;
  width: 50px;
}

.Project-info {
 width: 50%;
 text-align: left;
 padding: 20px;
}

.Project-links a {
  font-size: 2.5rem;
  padding: 10px;
}

.demo {
  color: rgb(87, 177, 250);
}

.github {
  color: black;
}

.Project-img {
  width: 40%;
  padding: 20px;
  margin: auto;
}

.Project-img img {
  height: 70%;
  width: 100%;
  border-radius: 10px;
  border: 2px solid;
  box-shadow: 0px 0px 50px 5px;
}

.reptifeed {
  background-color: rgba(69, 153, 0, 0.35);
  box-shadow: 0px 0px 20px 15px rgba(69, 153, 0, 0.5),
    inset 0px 0px 10px 5px rgba(69, 153, 0, 0.5);
}

.reptifeed img {
  border-color: rgba(154, 177, 6);
}

.eight-bit-clan {
  background-color: rgba(250, 250, 250, 0.35);
  color: rgb(12, 17, 88);
  box-shadow: 0px 0px 20px 15px rgba(250, 250, 250, 0.5),
    inset 0px 0px 10px 5px rgba(250, 250, 250, 0.5);
}
  
.eight-bit-clan img {
  border-color: rgb(37, 53, 113);
}
  
.catch-your-feelings {
  background-color: rgba(156, 156, 201, 0.5);
  box-shadow: 0px 0px 20px 15px rgb(156, 156, 201, 0.5),
    inset 0px 0px 10px 5px rgb(156, 156, 201, 0.5);
}
  
.catch-your-feelings img {
  border-color: rgb(56, 44, 44);
}
    
.musophile {
  background-color: rgba(98, 187, 255, 0.35);
  box-shadow: 0px 0px 20px 15px rgba(98, 187, 255, 0.5),
    inset 0px 0px 10px 5px rgba(98, 187, 255, 0.5);
}

.musophile img {
  border-color: 2px solid rgb(182, 237, 255);
}

@media (max-width: 1200px) {
  .Project-info-img-container {
    flex-direction: column-reverse;
  }

  .Project-info {
    width: 96%;
    text-align: center;
  }
  
  .Project-info li {
    list-style-type: none;
    padding: 0;
    margin: 0;
  }

  .Project-img {
    width: 80%;
  }
}