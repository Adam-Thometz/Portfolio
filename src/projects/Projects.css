.Projects {
  margin: 32px auto;
  animation: slide-up var(--duration) calc(var(--delay) * var(--i)) ease-in-out
    both;
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  transition: opacity 0.3s ease-in-out;
  position: relative;
}

.Projects-project-list {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin: 0 auto;
}

.project {
  transition: opacity 0.3s ease-in-out;
}

.display {
  transition: opacity 0.3s ease-in-out;
}

/* Styles for projects */

.art-buddy {
  background-color: rgba(77, 79, 94, 0.35);
  color: rgb(137, 235, 10);
  box-shadow: 0px 0px 20px 15px rgba(253, 77, 58, 0.5);
}

.art-buddy .Project-img img {
  border-color: rgb(255, 232, 26);
}

.reptifeed {
  background-color: rgba(69, 153, 0, 0.35);
  color: #d5ce3b;
  box-shadow: 0px 0px 20px 15px rgba(69, 153, 0, 0.5);
}

.eight-bit-clan {
  color: rgba(250, 250, 250);
  background-color: rgb(12, 17, 88, 0.35);
  box-shadow: 0px 0px 20px 15px rgba(250, 250, 250, 0.5);
}

.vesta-healing-psychotherapy {
  background-color: rgba(203, 155, 255, 0.35);
  color: rgba(255, 197, 162, 1);
  box-shadow: 0px 0px 20px 15px rgba(245, 129, 139, 0.5);
}

@media (max-width: 550px) {
  .Projects {
    flex-direction: column;
    align-items: center;
  }
}
