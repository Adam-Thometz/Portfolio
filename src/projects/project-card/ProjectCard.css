.ProjectCard {
  position: relative;
  width: 35%;
  border-radius: 10px;
  cursor: pointer;
  margin: 5%;
  transform: none;
  transition: transform 0.3s ease-in-out;
  border: none;
  font-family: "Montserrat", sans-serif;
}

.ProjectCard img {
  height: 100%;
  width: 100%;
  border-radius: 10px;
  opacity: 1; 
  transition: opacity 0.3s ease-in-out;
}

.ProjectCard-name {
  position: absolute;
  z-index: 1;
  top: 50%;
  left: 50%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  transform: translate(-50%, -50%);
  opacity: 0;
  height: 100%;
  width: 100%;
  border-radius: 5px;
  transition: opacity 0.3s ease-in-out;
}

.ProjectCard:hover,
.ProjectCard:focus {
  transform: scale(1.1);
}

.ProjectCard:hover img,
.ProjectCard:focus img {
  opacity: 0.3;
}

.ProjectCard-name:hover {
  opacity: 1;
}

.ProjectCard:focus .ProjectCard-name {
  opacity: 1;
}

@media (max-width: 550px) {
  .ProjectCard {
    width: 70%;
  }
}
