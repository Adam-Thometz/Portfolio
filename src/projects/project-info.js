// Project pictures
import artBuddy from './screenshots/art-buddy.png';
import reptifeed from './screenshots/reptifeed.png';
import bitClan from './screenshots/8-bit-clan.png';
import catchYourFeelings from './screenshots/catch-your-feelings.png';
import musophile from './screenshots/musophile.png';

// Awards
import mintbeanStar from './screenshots/mintbean-star.gif';

const projects = [
  {
    name: 'Art Buddy',
    description: 'First "real-world" project since graduating from bootcamp. Art Buddy is a collection of arts-based games and tools designed specifically for the special education classroom and reaching student academic goals. Games are based on real lessons I taught during my time as a special education music teacher. Art Buddy was made in collaboration with a UX designer and fellow educator. This game is currently in progress. For now, the only working games are Word-To-Music Decoder, Sequence Maker, and Score Keeper. There are remnants of Whats That Instrument currently deployed.',
    techStack: ["Javascript", "React", "CSS", "Redux", "Jest"],
    img: artBuddy,
    links: {
      demo: 'https://art-buddy.surge.sh',
      github: 'https://github.com/Adam-Thometz/Art-Buddy'
    }
  },
  {
    name: 'Reptifeed',
    description: 'My second and final capstone project for Springboard. Inspired by my new baby blue-tongue skink, Unity, this app gives you ideas on what to feed your reptile and helps you keep track of the foods you have for them. Features a homemade API for food information.',
    techStack: ["Javascript", "Node", "Express", "React", "CSS", "SQL", "Postgres", "Jest"],
    login: {
      username: 'abc',
      password: '123'
    },
    img: reptifeed,
    links: {
      demo: 'https://reptifeed.surge.sh',
      github: 'https://github.com/Adam-Thometz/Reptifeed'
    }
  },
  {
    name: '8-Bit Clan',
    award: mintbeanStar,
    description: "A pixel art maker created for the Mintbean Hiring Hackathon, Nov 2021. Create art, save it to your profile, and see if it made it to the trending list! Built in a week with UX designer Melody Acosta. Winner of Mintbean Star, awarded to top 5% of competitors.",
    techStack: ["Javascript", "Node", "Express", "React", "JSS", "SQL", "Postgres", "Heroku"],
    login: {
      username: 'abc',
      password: '123'
    },
    img: bitClan,
    links: {
      demo: 'https://8-bit-clan.surge.sh/',
      github: 'https://github.com/Adam-Thometz/8-Bit-Clan'
    }
  },
  {
    name: 'Catch Your Feelings',
    description: "An app that I invented for my day job as a music teacher for autistic students in grades PreK-5. You describe how happy and energetic you feel using icons and you get a song back based on the input. This project is currently being revamped!",
    techStack: ["Python", "Flask", "HTML", "CSS", "Javascript", "Heroku"],
    img: catchYourFeelings,
    links: {
      demo: 'https://catch-your-feelings.herokuapp.com/',
      github: 'https://github.com/Adam-Thometz/Catch-Your-Feelings/tree/flask'
    }
  },
  // {
  //   name: "Colorful Melodies",
  //   description: "An app that takes words and turns them into a simple melody by having students identify letters. Used as a tool in my music classroom to teach musical pitch by associating pitches with colors and letters. Created with the goal of incorporating letter recognition into mutlisensory musical instruction for students with learning and sensory-based disabilities. Uses the Web Audio API.",
  //   techStack: ["HTML", "CSS", "Javascript"],
  //   img: musicColors,
  //   links: {
  //     demo: '',
  //     github: ''
  //   }
  // },
  {
    name: 'Musophile',
    description: 'My first capstone project for Springboard. A website where you can curate your own music library. Created using the Spotify and MusicBrainz APIs.',
    techStack: ["Python", "Flask", "Postgres", "HTML", "CSS", "Javascript", "jQuery", "Heroku"],
    login: {
      username: 'abc',
      password: '12345678'
    },
    img: musophile,
    links: {
      demo: 'https://musophile.herokuapp.com/',
      github: 'https://github.com/Adam-Thometz/Musophile'
    }
  },
];

export default projects;