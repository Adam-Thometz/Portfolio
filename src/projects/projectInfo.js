// Project pictures
import artBuddy from '../_media/imgs/art-buddy.webp';
import reptifeed from '../_media/imgs/reptifeed.webp';
import bitClan from '../_media/imgs/8-bit-clan.webp';
import musophile from '../_media/imgs/musophile.webp';

// Awards
import mintbeanStar from '../_media/imgs/mintbean-star.gif';

const projects = [
  {
    name: 'Art Buddy',
    shortDescription: 'Arts-based games for special education',
    description: <><p>Art Buddy is a collection of arts-based games and tools designed specifically for the special education classroom and reaching students' academic goals. I worked with UX designer/fellow educator <a href='https://uxfol.io/mattiebecker'>Mattie Becker</a> to make this.</p><b>NOTE: I recommend viewing on a desktop. This app was purposely designed for a large screen.</b></>,
    learnedHowTo: [
      'make my code readable, obvious, and testable (I call it "making my code ROT")',
      'use package.json scripts to minimize typing in the terminal',
      'make music play in the browser depending on app settings',
      'create a component library of common UI elements',
      'articulate my vision for a product',
      'incorporate user feedback',
    ],
    techStack: ["Javascript", "React", "CSS", "Redux", "Jest"],
    img: artBuddy,
    links: {
      demo: 'https://art-buddy.surge.sh',
      github: 'https://github.com/Adam-Thometz/Art-Buddy'
    }
  },
  {
    name: 'Reptifeed',
    shortDescription: 'A reptilian diet tracker',
    description: <p>My second and final capstone project for Springboard. Inspired by my blue-tongue skink, Unity, this app gives you ideas on what to feed your reptile and helps you keep track of the foods you have for them.</p>,
    learnedHowTo: [
      'build admin-only features',
      'make a website/app accessible',
      'make a responsive website',
      'invent my own API'
    ],
    techStack: ["Javascript", "Node", "Express", "React", "CSS", "Postgres", "Jest"],
    login: {
      username: 'abc',
      password: '123'
    },
    img: reptifeed,
    links: {
      demo: 'https://reptifeed.surge.sh',
      github: 'https://github.com/Adam-Thometz/Reptifeed'
    }
  },
  {
    name: '8-Bit Clan',
    award: mintbeanStar,
    shortDescription: 'An 8-bit NFT maker',
    description: <p>A pixel art maker created for the Mintbean Hiring Hackathon, Nov 2021. Create art, save it to your profile, and see if it made it to the trending list! Built with UX designer <a href='https://melodyux.design/'>Melody Acosta</a> in a week. Winner of Mintbean Star, awarded to top 5% of competitors.</p>,
    learnedHowTo: [
      'collaborate with a UX designer',
      'work efficiently under tight time constraints',
      'store art data in a database and call on that data to reproduce a smaller version of that art'
    ],
    techStack: ["Javascript", "Node", "Express", "React", "JSS", "Postgres"],
    login: {
      username: 'abc',
      password: '123'
    },
    img: bitClan,
    links: {
      demo: 'https://8-bit-clan.surge.sh/',
      github: 'https://github.com/Adam-Thometz/8-Bit-Clan'
    }
  },
  {
    name: 'Musophile',
    shortDescription: 'An obscure music curator',
    description: <p>My first capstone project for Springboard. A website where you can curate your own music library. Created using the Spotify and MusicBrainz APIs.</p>,
    learnedHowTo: [
      'connect frontend and backend using server-side rendering',
      'use the Spotify API and OAuth'
    ],
    techStack: ["Python", "Flask", "Postgres", "HTML", "CSS", "Javascript", "jQuery", "Bootstrap"],
    login: {
      username: 'abc',
      password: '12345678'
    },
    img: musophile,
    links: {
      demo: 'https://musophile.herokuapp.com/',
      github: 'https://github.com/Adam-Thometz/Musophile'
    }
  },
];

export default projects