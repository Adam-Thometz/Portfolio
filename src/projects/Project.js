import React from "react";
import { Link } from "react-router-dom";
import TechStack from "../tech-stack/TechStack";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faGlobe } from "@fortawesome/free-solid-svg-icons";
import { faGithub } from "@fortawesome/free-brands-svg-icons"
import './Project.css';

const Project = ({name, award, description, techStack, login, img, links}) => {
  const cssClass = name.toLowerCase().replaceAll(' ', '-').replace('8', 'eight');

  return (
    <div className={`Project ${cssClass}`} key={cssClass}>
      <h2>{name}</h2>
      <div className="Project-info-img-container">
        <div className="Project-info" tabIndex={0}>
          <div className="Project-description">
            <h3>Description</h3>
            <p>{description}</p>
            {award ? <img className="Project-award" src={award} alt={`An award for ${name}`} /> : null}
          </div>
          <div className="Project-links">
            <Link className="demo" to="" onClick={() => window.open(links.demo)}><FontAwesomeIcon title={`Demo for ${name}`} icon={faGlobe} /></Link>
            <Link className="github" to="" onClick={() => window.open(links.github)}><FontAwesomeIcon title={`Github repo for ${name}`} icon={faGithub} /></Link>

            {login ? (
              <div>
                <b>To get the most out of this site, log in with the following:</b>
                <ul>
                  <li>Username: {login.username}</li>
                  <li>Password: {login.password}</li>
                </ul>
              </div>
            ) : null}
          </div>
        </div>
        <div className="Project-img">
          <img src={img} alt={`A screenshot of ${name}`} />
        </div>
      </div>
      <TechStack techStack={techStack} />
    </div>
  );
};

export default Project;