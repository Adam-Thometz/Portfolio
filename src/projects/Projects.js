import React from "react";
import projects from "./project-info";
import Project from "./Project";
import NavContact from "../navbar-contact/NavContact";
import './Projects.css'

const Projects = () => {
  return (
    <main className="Projects">
      <h1 className="Projects-header">Projects</h1>
      {projects.map(p => (
        <Project name={p.name} award={p.award} description={p.description} techStack={p.techStack} login={p.login} img={p.img} links={p.links} />
        ))}
      <NavContact />
    </main>
  );
};

export default Projects;