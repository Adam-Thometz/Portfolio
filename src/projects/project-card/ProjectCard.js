import React from 'react';

import './ProjectCard.css';

import Project from '../project/Project';

import projects from '../../_data/projectInfo';

const ProjectCard = ({ name, img, shortDescription, setShownProject }) => {
  const cssClass = name.toLowerCase().replaceAll(' ', '-').replace('8', 'eight');
  const handleSetShownProject = () => {
    const {
      award,
      description,
      learnedHowTo,
      techStack,
      login,
      links
    } = projects.find(p => p.name === name);
    setShownProject(<Project
      name={name}
      img={img}
      award={award}
      description={description}
      learnedHowTo={learnedHowTo}
      techStack={techStack}
      login={login}
      links={links}
      setShownProject={setShownProject}
    />);
  };
  return (
    <figure className={`ProjectCard ${cssClass}`} onClick={handleSetShownProject}>
      <figcaption>
        <h2>{name}</h2>
        <p>{shortDescription}</p>
      </figcaption>
      <img src={img} alt={name} />
    </figure>
  )
}

export default ProjectCard