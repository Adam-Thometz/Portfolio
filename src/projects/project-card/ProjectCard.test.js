import React from "react";
import { render } from "@testing-library/react";
import ProjectCard from "./ProjectCard";

import projects from "../projectInfo";

describe("ProjectCard component", () => {
  it("renders without crashing", () => {
    render(<ProjectCard project={projects[0]} />);
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(<ProjectCard project={projects[0]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
