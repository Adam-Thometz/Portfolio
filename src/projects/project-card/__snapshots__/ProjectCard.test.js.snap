// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard component matches the snapshot 1`] = `
<DocumentFragment>
  <button
    class="ProjectCard art-buddy"
    tabindex="0"
  >
    <div
      class="ProjectCard-name"
    >
      <h2>
        Art Buddy
      </h2>
      <p>
        Arts-based games for special education
      </p>
    </div>
    <img
      alt="Art Buddy"
      src="art-buddy.webp"
    />
  </button>
</DocumentFragment>
`;
